sockeye/layers.py

```diff
@@ -117,6 +119,49 @@ def forward(self, data: pt.Tensor, vocab_slice_ids: Optional[pt.Tensor] = None)
         return F.linear(data, weight, bias)
 
 
+class KNN(pt.nn.Module):
+    def __init__(self, keys_index, vals, vocab_size, k=64, temperature=10, state_dump=None) -> None:
```

Comment:
can we add type annotations for the first 3 positional arguments?

Reply:
Looking closer, adding type annotation for `keys_index` will be tricky because it is supposed to be a `faiss.Index` type, but we want to do optional import for faiss so I don't think there is an easy way to specify it.

(I could do a comment though to circumvent the import.)

Reply:
You could use a [forward reference](https://peps.python.org/pep-0484/#forward-references) as type hint, i.e. `keys_index: 'faiss.Index'`

Reply:
Ah TIL :)